include/aspect/geometry_model/ellipsoidal_chunk.h

```diff
@@ -38,12 +40,77 @@ namespace aspect
     using namespace dealii;
 
     /**
+     * Define a type to know what type of topography we
+     * are using
+     */
+    enum topo_types {NO_TOPOGRAPHY,
+                     PRM_EXACT,
+                     PRM_UNIFORM_GRID_INTERPOLATED,
+                     FILE_UNIFORM_GRID,
+                     FILE_NONUNIFORM_GRID
+                    };
+
+    /**
      * A class that describes a geometry for an ellipsoid such as the WGS84 model of the earth.
      */
     template <int dim>
     class EllipsoidalChunk : public Interface<dim>
     {
       public:
+        static bool In2dPolygon(dealii::Point<2> &point,const std::vector<std::vector<double> > &pointList);
```

Comment:
Btw, such functions are often best placed into some anonymous namespace in the .cc file, since I assume you don't want to use it from another class.